File: BookExamples/47_TabLayoutDemoV2/app/src/main/java/com/scottshaper/tablayoutdemov2/ThirdFragment.java
package com.scottshaper.tablayoutdemov2;

import android.net.Uri;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

public class ThirdFragment extends Fragment {

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState
    ) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_thrid, container, false);
    }

    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
    }
    //I HAD TO ADD THIS
    public interface OnFragmentInteractionListener {
        // TODO: Update argument type and name
        void onFragmentInteraction(Uri uri);
    }
}

File: MultiFragmentNavigation/app/src/main/java/com/example/multifragmentnavigation/ThirdFragment.kt
package com.example.multifragmentnavigation

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.net.Uri
import com.example.multifragmentnavigation.databinding.FragmentThirdBinding



class ThirdFragment : Fragment() {
    private var _binding: FragmentThirdBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        //REPLACED WITH BINDING
        // Inflate the layout for this fragment
        //return inflater.inflate(R.layout.fragment_second, container, false)
        _binding = FragmentThirdBinding.inflate(inflater, container, false)
        return binding.root
    }

    interface OnFragmentInteractionListener{
        fun onFragment3Interaction(uri: Uri)
    }

    override fun onStart(){
        super.onStart()
        arguments?.let {
            val args = ThirdFragmentArgs.fromBundle(it)
                binding.textViewName.text = args.message1
            }
    }


}